### DISCUSSION TITLE
[Fast Data] Set a single-view expiration date

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#307] [Ideas]
### Which area of this project could be most improved?

Fast Data

### Description

Hi guys,
recently, for a client, we had to implement a new (to us at least) fast-data use case.

The single-views generation is triggered by projection changes, that are generated after a new projection update.

In our case, we need to generate a single view after an expiration date.
Our single view has a range of times in which it is valid. These ranges are stored in projections that we can receive a lot of time before their start date.
When the old rage expires, the single views pointing to that range must be generated again, but without a projection update.
Can we add an expiration field on the single-view, or something like that, to have another trigger that the SVC should check to generate a single-view?

**Comment by ianFar96** [comment]
Hi @silversoul93 If I understood correctly you need an automatic trigger to re-aggregate a Singe View based on a timestamp or a TTL index on the Single View itself, is that correct? If that's the case the Fast Data reactivity from Projection to Single View looses a bit its meaning. Are you planning on disabling the Strategy execution / Projection Changes alltogether?
